docs/decisions/0019-oep-58-atlas-translations-design.rst

```diff
@@ -0,0 +1,423 @@
+
+The ``get_python_locale_directory`` will support two modes:
+
+#. When ``ENABLE_ATLAS_TRANSLATIONS`` is disabled, the XBlock translations
+   from the ``pip`` packages will be used such as the
+   ``lib/python3.8/site-packages/drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po``
+   path for the Drag and Drop XBlock.
+
+#. When ``ENABLE_ATLAS_TRANSLATIONS`` is enabled, the atlas translations will
+   be used which is located in the ``edx-platform`` in an the git-ignored
+   ``edx-platform/conf/plugins-locale/drag_and_drop_v2/ar/LC_MESSAGES/text.po``
+   path. This file pulled by the ``pull_plugins_translations`` command.
+
+
+XBlockI18nService support for ``atlas`` JavaScript translations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``XBlockI18nService`` will provide a new centralized
+``get_javascript_locale_path`` method to get the JavaScript translations
+``django.js`` file.
+
+This function needs to be used by the XBlocks in an opt-in backward-compatible
+manner.
+
+A new ``i18n_js_namespace`` property is needed for the :ref:`compile-js-command`
+to generate JavaScript translations in a centrally managed manner for all
+XBlocks as described in the :ref:`js-translations` section.
+
+The ``i18n_js_namespace`` property will eliminate the need to hardcode the
+namespace the `XBlock Makefile compile_translations rule`_.
+
+
+For example, the `Drag and Drop XBlock get_static_i18n_js_url`_ will need to
+be updated to support both the ``XBlockI18nService`` new
+``get_javascript_locale_path`` method and the namespace.
+
+.. code:: diff
+
+     class DragAndDropBlock(XBlock):
+
+   +   i18n_js_namespace = 'DragAndDropI18N'
+
+       @staticmethod
+       def _get_statici18n_js_url():
+           """
+           Returns the Javascript translation file for the currently selected language, if any found by
+           `pkg_resources`
+           """
+           lang_code = translation.get_language()
+           if not lang_code:
+               return None
+
+   +       # TODO: Make this the default once OEP-58 is implemented.
+   +       if hasattr(self.i18n_service, 'get_javascript_locale_path'):
+   +           atlas_locale_path = self.i18n_service.get_javascript_locale_path()
+   +           if atlas_locale_path:
+   +               return atlas_locale_path
+
+           text_js = 'public/js/translations/{lang_code}/text.js'
+           country_code = lang_code.split('-')[0]
+           for code in (translation.to_locale(lang_code), lang_code, country_code):
+               if pkg_resources.resource_exists(loader.module_name, text_js.format(lang_code=code)):
+                   return text_js.format(lang_code=code)
+           return None
+
+
+New ``compile_plugins_js_translations`` command
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This command will loop over XBlock modules that has the ``i18n_js_namespace``
+property and compile the JavaScript translations.
```

Comment:
> and compile the JavaScript translations.

It's not clear from this documentation what compiling js translations implies. I'm not sure if information explaining that should live here, or if we should link out to something else that explains it, but it'd be nice to make that a little more clear somehow.

Reply:
I've linked to the [`compilejsi18n` command docs](https://django-statici18n.readthedocs.io/en/latest/commands.html#compilejsi18n).